src/groups/wrappers/z_bmq/z_bmqt/z_bmqt_compressionalgorithmtype.h

```diff
@@ -0,0 +1,73 @@
+#ifndef INCLUDED_Z_BMQT_COMPRESSIONALGORITHMTYPE
+#define INCLUDED_Z_BMQT_COMPRESSIONALGORITHMTYPE
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+#include <stdbool.h>
+
+struct z_bmqt_CompressionAlgorithmType {
```

Comment:
This uses C++ constructs, it should probably just be rexposed as an enum:

```c
typedef enum {
  z_bmqt_CompresssionAlgorithmType__e_UKNOWN = -1,
  ...
} z_bmqt_CompressionAlgorithmType;
```

Unfortunately, there's no real easy way to define this enum in terms of its C++ equivalent, as that would require accessing a C++ symbol in the header file. There are probably some tricks, but I don't think it would be worth it. What we can do is add some checks at compile time that our C header definitions match the C++ definitions with a static assert:

```c++
// in z_bmqt_compressionalgorithmtype.cpp
#include <z_bmqt_compressionalgorithmtype.h>

#include <bmqt_compressionalgorithmtype.h>
#include <bslmf_assert.h>


BSLMF_ASSERT(z_bmqt_CompressionAlgorithmType__e_UNKNOWN == bmqt::CompressionAlgorithmType::e_UNKNOWN);
...
```